rules/import-style.js

```diff
@@ -0,0 +1,316 @@
+			if (allowedImportStyles.get('default') && !allowedImportStyles.get('namespace')) {
+				effectiveAllowedImportStyles = new Map(allowedImportStyles);
+				effectiveAllowedImportStyles.set('namespace', true);
+			}
+		}
+
+		if (actualImportStyles.every(style => effectiveAllowedImportStyles.get(style))) {
+			return;
+		}
+
+		const data = {
+			allowedStyles: joinOr([...allowedImportStyles.keys()]),
+			moduleName
+		};
+
+		context.report({
+			node,
+			messageId: MESSAGE_ID,
+			data
+		});
+	};
+
+	let visitor = {};
+
+	if (checkImport) {
+		visitor = {
+			...visitor,
+
+			ImportDeclaration(node) {
+				const moduleName = getStringIfConstant(node.source, context.getScope());
+
+				const allowedImportStyles = styles.get(moduleName);
+				const actualImportStyles = getActualImportDeclarationStyles(node);
+
+				report(node, moduleName, actualImportStyles, allowedImportStyles);
+			}
+		};
+	}
+
+	if (checkDynamicImport) {
+		visitor = {
+			...visitor,
+
+			'ExpressionStatement > ImportExpression'(node) {
+				const moduleName = getStringIfConstant(node.source, context.getScope());
+				const allowedImportStyles = styles.get(moduleName);
+				const actualImportStyles = ['unassigned'];
+
+				report(node, moduleName, actualImportStyles, allowedImportStyles);
+			},
+
+			[assignedDynamicImportTemplate](node) {
+				const assignmentTargetNode = assignedDynamicImportTemplate.context.getMatch(assignmentTargetVariable);
+				const moduleNameNode = assignedDynamicImportTemplate.context.getMatch(moduleNameVariable);
+				const moduleName = getStringIfConstant(moduleNameNode, context.getScope());
+
+				if (!moduleName) {
+					return;
+				}
+
+				const allowedImportStyles = styles.get(moduleName);
+				const actualImportStyles = getActualAssignmentTargetImportStyles(assignmentTargetNode);
+
+				report(node, moduleName, actualImportStyles, allowedImportStyles);
+			}
+		};
+	}
+
+	if (checkRequire) {
+		visitor = {
+			...visitor,
+
+			'ExpressionStatement > CallExpression[callee.name=\'require\'][arguments.length=1]'(node) {
+				const moduleName = getStringIfConstant(node.arguments[0], context.getScope());
+				const allowedImportStyles = styles.get(moduleName);
+				const actualImportStyles = ['unassigned'];
+
+				report(node, moduleName, actualImportStyles, allowedImportStyles, true);
+			},
+
+			[assignedRequireTemplate](node) {
+				const assignmentTargetNode = assignedRequireTemplate.context.getMatch(assignmentTargetVariable);
+				const moduleNameNode = assignedRequireTemplate.context.getMatch(moduleNameVariable);
+				const moduleName = getStringIfConstant(moduleNameNode, context.getScope());
+
+				if (!moduleName) {
+					return;
+				}
+
+				const allowedImportStyles = styles.get(moduleName);
+				const actualImportStyles = getActualAssignmentTargetImportStyles(assignmentTargetNode);
+
+				report(node, moduleName, actualImportStyles, allowedImportStyles, true);
+			}
+		};
+	}
+
+	return templates.visitor(visitor);
```

Comment:
I haven't see  `templates.visitor`, should it skip if `vistor` is  empty?

Reply:
I don't think performance gain here is worth the code complication.